lexer.py

```diff
@@ -0,0 +1,127 @@
+from tokens import TokenTypes, Token
+
+###############################################################################
+#                                                                             #
+#   LEXER                                                                     #
+#                                                                             #
+###############################################################################
+
+ESCCHARS = {
+    "n":  "\n",
+    "t":  "\t",
+    "\\": "\\",
+    "'": "'",
+    "r":  "\r",
+}
+
+RESERVED_WORDS = {
+            "SELECT",
+            "FROM",
+            "EDGES",
+            "NODES",
+            "TRAVERSE",
+            "BY",
+            "DEPTH",
+            "BREADTH",
+            "WHERE",
+            "LIMIT",
+}
+
+class Lexer(object):
+    def __init__(self, text):
+        self.text = text         # string input
+        self.pos = 0             # self.pos is index into self.text
+        self.line = 1            # current line
+        self.current_char = self.text[0] # current char
+        self.tokens = []         # tokens so far
+    def _error(self, msg=""):
```

Comment:
I suggest removing the default value since you don't use it, and we probably always want to be providing a specific error message.